app/uk/gov/hmrc/teamsandrepositories/models/BuildJob.scala

```diff
@@ -22,54 +22,96 @@ import uk.gov.hmrc.mongo.play.json.formats.MongoJavatimeFormats
 
 import java.time.Instant
 
-case class BuildJob(
-  service    : String,
-  jenkinsURL : String,
-  latestBuild: Option[BuildJobBuildData]
-)
+case class BuildData(
+                      number: Int,
+                      url: String,
+                      timestamp: Instant,
+                      result: Option[BuildResult]
+                    )
+object BuildData {
+
+  val mongoFormat: OFormat[BuildData] =
+    ((__ \ "number").format[Int]
+      ~ (__ \ "url").format[String]
+      ~ (__ \ "timestamp").format(MongoJavatimeFormats.instantFormat)
+      ~ (__ \ "result").formatNullable[BuildResult]
+      ) (apply, unlift(unapply))
+
+  val apiWrites: Writes[BuildData] =
+    ((__ \ "number").write[Int]
+      ~ (__ \ "url").write[String]
+      ~ (__ \ "timestamp").write[Instant]
+      ~ (__ \ "result").writeNullable[BuildResult]
+      ) (unlift(unapply))
+
+  val jenkinsReads: Reads[BuildData] =
+    ((__ \ "number").read[Int]
+      ~ (__ \ "url").read[String]
+      ~ (__ \ "timestamp").read[Instant]
+      ~ (__ \ "result").readNullable[BuildResult]
+      ) (apply _)
+}
+
+
+sealed trait JenkinsObject
+
+case class JenkinsFolder(service: String, jenkinsURL: String, objects: Seq[JenkinsObject]) extends JenkinsObject
+case class BuildJob(service: String, jenkinsURL: String, latestBuild: Option[BuildData]) extends JenkinsObject
+
+case class PipelineJob(service: String, jenkinsURL: String) extends JenkinsObject
 
 object BuildJob {
 
   val mongoReadFormat: Reads[BuildJob] =
     ((__ \ "service").read[String]
       ~ (__ \ "jenkinsURL").read[String]
-      ~ (__ \ "latestBuild").readNullable(BuildJobBuildData.mongoFormat.reads)
+      ~ (__ \ "latestBuild").readNullable(BuildData.mongoFormat.reads)
       ) (apply _)
 
   val mongoWriteFormat: Writes[BuildJob] =
     ((__ \ "service").write[String]
       ~ (__ \ "jenkinsURL").write[String]
-      ~ (__ \ "latestBuild").writeNullable(BuildJobBuildData.mongoFormat.writes)
+      ~ (__ \ "latestBuild").writeNullable(BuildData.mongoFormat.writes)
       ) (unlift(unapply))
 
   val mongoFormat: Format[BuildJob] = Format(mongoReadFormat, mongoWriteFormat)
```

Comment:
we could just define mongoFormat as `(__ \ "service").format[String] etc` rather than Reads and Writes individually.
`BuildData.mongoFormat` could be provided implicitly in the definition scope. Note `.writeNullable[BuildData]` i.e. providing the type, would be more explicit either way